crates/provider/src/blocks.rs

```diff
@@ -123,13 +133,17 @@ impl<N: Network> NewBlocks<N> {
                 yield known_block;
             }
 
+            // If we're paused, wait until we're unpaused.
+            // Once unpaused, reset `self.next_yield` to ignore the blocks that were included while we were paused.
+            let unpaused = self.check_paused().await;
```

Comment:
this is a bit problematic because we have different stream types

if we stop polling the subscription, we will just accumulate a bunch of messages is buffers

Reply:
right, I would've guessed they would just get discarded after filling the buffer?